test/Models/createCatalogItemFromUrlSpec.ts

```diff
@@ -0,0 +1,54 @@
+import Terria from "../../lib/Models/Terria";
+import createCatalogItemFromUrl from "../../lib/Models/createCatalogItemFromUrl";
+import WebMapServiceCatalogGroup from "../../lib/Models/WebMapServiceCatalogGroup";
+import GeoJsonCatalogItem from "../../lib/Models/GeoJsonCatalogItem";
+import CatalogMemberFactory from "../../lib/Models/CatalogMemberFactory";
+
+describe("createCatalogItemFromUrl", function() {
+  let terria: Terria;
+
+  beforeEach(function() {
+    terria = new Terria();
+
+    CatalogMemberFactory.register(
+      WebMapServiceCatalogGroup.type,
+      WebMapServiceCatalogGroup
+    );
+    CatalogMemberFactory.register(GeoJsonCatalogItem.type, GeoJsonCatalogItem);
+
+    createCatalogItemFromUrl.register(
+      s => true,
+      WebMapServiceCatalogGroup.type,
+      true
+    );
+    createCatalogItemFromUrl.register(
+      s => Boolean(s.match(new RegExp("\\.geojson$", "i"))),
```

Comment:
Just pull in the `registerCatalogMembers` from `registerCatalogMembers.ts` & use that, OR make `matchesExtension` from `createCatalogItemFromUrl.ts` an export & use that